acstools/findsat_mrt.py

```diff
@@ -0,0 +1,1409 @@
+                                        update_dq)
+
+# test for matplotlib, turn off plotting if it does not exist
+try:
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+except ImportError:
+    plt = None
+    warnings.warn('matplotlib not found, plotting is disabled')
+
+__taskname__ = "findsat_mrt"
+__author__ = "David V. Stark"
+__version__ = "1.0"
+__vdate__ = "10-Feb-2023"
+__all__ = ['TrailFinder', 'WfcWrapper']
+
+# Initialize the logger
+logging.basicConfig()
+LOG = logging.getLogger(f'{__taskname__}')
+LOG.setLevel(logging.INFO)
+
+
+class TrailFinder:
+    '''Top-level class to handle trail identification and masking.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    header : `astropy.io.fits.Header`, optional
+        The primary header for the input data (usually the 0th extension). This
+        is not used for anything during the analysis, but it is saved with the
+        output mask and satellite trail catalog so information about the
+        original observation can be easily retrieved. Default is `None`.
+    image_header : `astropy.io.fits.Header`, optional
+        The specific header for the FITS extension being used. This is
+        added onto the catalog. Default is `None`.
+    save_image_header_keys : list, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.save_image_header_keys`.
+        Default is an empty list (nothing saved from ``image_header``).
+    processes : int, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.processes`.
+        The default is 2.
+    min_length : int, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.min_length`.
+        The default is 25 pixels.
+    max_width : int, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.max_width`.
+        The default is 75 pixels.
+    buffer : int, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.buffer`.
+        The default is 250 pixels on each side.
+    threshold : float, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.threshold`.
+        The default is 5.
+    theta : ndarray, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.theta`.
+        The default is `None`, which sets to ``numpy.arange(0, 180, 0.5)``.
+    kernels : list, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.kernels`.
+        The default is `None`, which reverts to using the 3-, 7-, and 15-pixel
+        wide kernels included with this package.
+    mask_include_status : list, optional
+        See :attr:`~acstools.utils_findsat_mrt.TrailFinder.mask_include_status`.
+        The default is ``[2]``.
+    plot : bool, optional
+        Plot all intermediate steps. When set, plots of the input image, MRT
+        with identified sources, and resulting image maks will be generated
+        after running :func:`acstools.findat_mrt.TrailFinder.__init__`:,
```

Comment:
`__init__` is not exposed in API doc.
```suggestion
        after running the constructor,
```